core/chains/evm/logpoller/log_poller.go

```diff
@@ -57,15 +59,23 @@ type logPoller struct {
 	backfillBatchSize int64         // batch size to use when backfilling finalized logs
 
 	filterMu  sync.Mutex
-	addresses map[common.Address]struct{}
-	eventSigs map[common.Hash]struct{}
+	events    map[EventID]struct{}
+	callbacks map[EventID]Callback
 
 	replay chan int64
 	ctx    context.Context
 	cancel context.CancelFunc
 	done   chan struct{}
 }
 
+// NewLogPoller creates a log poller. Note there is an assumption
+// that blocks can be processed faster than they are produced for the given chain, or the poller will fall behind.
+// Block processing involves in the steady state (non-reorg case):
+// - eth_getBlockByNumber,
```

Comment:
Maybe worth noting something like:
```suggestion
// - eth_getBlockByNumber - headers only (transaction hashes, not full transaction objects),
```